### PR TITLE
FEATURE : Package controllers

### PR DESCRIPTION
**Pull request by PiRB**

### CODE REVIEW

File: enfivy-back/src/main/java/com/envify/back/controller/PackageVersionController.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    public void deletePackageById(@PathVariable int id) {
+        packageVersionService.deletePackageVersionById(id);
+    }
+
+    private void mapPackageVersionDtoToPackageVersionEntity(PackageVersionDto packageVersionDto, PackageVersionEntity packageVersionEntity, int packageVersionId, int packageId) {
+        packageVersionEntity.setId(packageVersionId);
+        packageVersionEntity.setVersionNumber(packageVersionDto.getVersionNumber());
+        packageVersionEntity.setUrl(packageVersionDto.getUrl());
+        packageVersionEntity.setVersionStatusId(packageVersionDto.getVersionStatusId());
+        packageVersionEntity.setPackageId(packageId);
```
Comment (Sem-Lord) [inline enfivy-back/src/main/java/com/envify/back/controller/PackageVersionController.java]: Will be cool if you can use a mapper. EX: mapstruct can do the job

File: enfivy-back/src/main/java/com/envify/back/dto/PackageDto.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.envify.back.dto;
+
+import java.util.Objects;
+
+public class PackageDto {
+    private int id;
+    private String name;
+
+    public int getId() { return id; }
```
Comment (Sem-Lord) [inline enfivy-back/src/main/java/com/envify/back/dto/PackageDto.java]: you forget to format

File: enfivy-back/src/main/java/com/envify/back/service/impl/VersionStatusServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Autowired
+    private VersionStatusDao versionStatusDao;
+
+    @Override
+    public List<VersionStatusEntity> findAllVersionsStatuses() { return versionStatusDao.findAll(); }
+
+    @Override
+    public void saveVersionStatus(VersionStatusEntity versionStatus) { versionStatusDao.save(versionStatus); }
+}
```
Comment (Sem-Lord) [inline enfivy-back/src/main/java/com/envify/back/service/impl/VersionStatusServiceImpl.java:28]: you forget to format

File: enfivy-back/src/main/java/com/envify/back/controller/PackageVersionController.java (hunk 1 above)
Comment (PiRB) [reply enfivy-back/src/main/java/com/envify/back/controller/PackageVersionController.java]: tu as pas dis que tu ferais le mapper???

Comment (PiRB) [reply enfivy-back/src/main/java/com/envify/back/controller/PackageVersionController.java]: apres je peux toujours le faire

File: enfivy-back/src/main/java/com/envify/back/service/mapper/PackageVersionMapper.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    PackageVersionEntity packageVersionDtoToPackageVersion(PackageVersionDto packageVersionDto);
+
+//    @Override
+//    public PackageVersionEntity packageVersionDtoToPackageVersion(PackageVersionDto packageVersionDto) {
+//        PackageVersionEntity packageVersionEntity = new PackageVersionEntity();
+//        packageVersionEntity.setVersionNumber(packageVersionDto.getVersionNumber());
+//        packageVersionEntity.setUrl(packageVersionDto.getUrl());
+//        packageVersionEntity.setVersionStatusId(packageVersionDto.getVersionStatusId());
+//
+//        return packageVersionEntity;
```
Comment (Sem-Lord) [inline enfivy-back/src/main/java/com/envify/back/service/mapper/PackageVersionMapper.java]: Remove comment

File: enfivy-back/src/main/java/com/envify/back/controller/PackageVersionController.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1/packages/{packageId}/versions")
+public class PackageVersionController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PackageVersionController.class);
+
+//    @Autowired
+//    private PackageVersionMapper packageVersionMapper;
```
Comment (Sem-Lord) [inline enfivy-back/src/main/java/com/envify/back/controller/PackageVersionController.java]: remove comment

**Review by Sem-Lord — APPROVED** [review]

**Pull request merged**